Scroller/EditableWrapper.js

```diff
@@ -95,27 +104,42 @@ const EditableWrapper = (props) => {
 	}, [dataSize]);
 
 	const startEditing = useCallback((item) => {
-		// FIXME: Need to figure out the right element
-		if (item.classList.contains('spottable')) {
+		if (item.dataset.index) {
+			Spotlight.set(mutableRef.current.spotlightId, {restrict: 'self-only'});
+
 			item.classList.add(css.selected);
+			item.classList.add(customCss.selected);
 			mutableRef.current.selectedItem = item;
 
 			mutableRef.current.fromIndex = Number(item.style.order) - 1;
 			mutableRef.current.prevToIndex = mutableRef.current.fromIndex;
 		}
-	}, []);
+	}, [customCss.selected]);
+
+	const findItemNode = useCallback((node) => {
+		for (let current = node; current !== scrollContentRef.current && current !== document; current = current.parentNode) {
+			if (current.dataset.index) {
+				return current;
+			}
+		}
+	}, [scrollContentRef]);
 
 	const handleClick = useCallback((ev) => {
+		const targetItemNode = findItemNode(ev.target);
+
 		if (mutableRef.current.selectedItem) {
 			// Finalize orders and forward `onComplete` event
 			const orders = finalizeOrders();
 			forwardCustom('onComplete', () => ({orders}))({}, editable);
 			reset();
-		} else {
+		} else if (targetItemNode && targetItemNode.dataset.index) {
 			// Start editing by adding selected transition to selected item
-			startEditing(ev.target.parentElement);
+			startEditing(targetItemNode);
 		}
-	}, [editable, finalizeOrders, reset, startEditing]);
+
+		ev.preventDefault();
```

Comment:
Would you please add some comments on why these are needed?

Reply:
done